### DISCUSSION TITLE
Proxy ports inside container

### DISCUSSION THREAD

**Discussion by giovanidiniz** [coder/coder#15726] [General] [answered]
I have a general question about how to best architect the system I'm building with coder. 

I am running a setup that has a server (running on port 9090) and a web UI (running on port 3080), both running inside the container workspace.

My current solution is to port both ports to my local machine, in the webUI I point the server address to localhost:9090 and bob's your uncle. 

I would like, however, to enable a cloud-only solution by exposing the webUI as a coder app and using the appropriate endpoint. I'm not entirely sure how to do that/where in the code to look at to modify to enable this functionality.

Any pointers would be much appreciated!

**Comment by matifali** [comment]
Hi @giovanidiniz, Have you looked at adding a [`coder_app`](https://registry.terraform.io/providers/coder/coder/latest/docs/resources/app) to your workspace template?

You can add a webUI `coder_app` as,

```tf
resource "coder_app" "webui" {
  agent_id     = coder_agent.dev.id
  slug         = "webui"
  display_name = "Web UI"
  url          = "http://localhost:3080"
  share        = "owner"
  subdomain    = true
}
```
This will add a button n your workspace page to access the webUI given your backend and frontend are running.

**Reply by giovanidiniz** [reply, marked as answer]
Thanks for the assist! I worked it out using port forwarding from the dashboard. super nice feature I completely overlooked.
